collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
```

Comment:
Daniel,

what about a change like this one?
http://linkode.org/ZRBW0unHfyJbllusyXJKe3/oV8avYUzsXR8Zsl1HzoVQ
This is basically the technique in the article to cope with unhashable
descriptor owners. Not sure if this is what you meant. I think we're almost
there, just need to iron out a few details to get this landed.

On Sun, Mar 1, 2015 at 9:10 PM, Daniel DeSousa <notifications@github.com>
wrote:

> In collection_json.py
> <https://github.com/ricardokirkner/collection-json.python/pull/12#discussion_r25573849>
> :
>
> > @@ -6,6 +6,59 @@
> >  __version__ = '0.1.0'
> >
> >
> > +class ArrayProperty(object):
> > +
> > +    """Property which converts from any enumerable to a typed Array instance."""
> > +
> > +    def __init__(self, cls, name):
> > +        """Constructs typed array property
> > +
> > +        :param cls type: the type of objects expected in the array
> > +        :param name str: name of the property
> > +        """
> > +        self.cls = cls
> > +        self.name = name
> > +        self.value_attr = "_" + name
>
> So I was excited to implement this when I saw it, as it looks like a nice
> solution to the problem. The post even describes using WeakRefDictionary,
> addressing my immediate concerns about memory leaks via cycle.
>
> Unfortunately I don't think the implementation would be sound in our case.
> For dictionaries, you need to implement *hash* and *eq* and preserve the
> property if eq(o1) == eq(o2), then hash(o1) == hash(o2). Since we override
> eq and ne, we must implement a decent non-trivial hash function.
>
> The bigger problem is that mutability would cause the hash in this
> function to change, meaning the next look-up could fail. Worse yet, if
> users created multiple objects with the same values such they were equal,
> it would also cause the instances to hash to the same value and change the
> property of both objects -- a consequence of aggregating the objects
> together at the class level. I think this probably wouldn't work in all but
> the most trivial cases.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ricardokirkner/collection-json.python/pull/12/files#r25573849>
> .
>